integration/benchmark/main.go

```diff
@@ -0,0 +1,234 @@
+	testing.Init()
+	// disable the automatic iteration count system
+	flag.Set("test.benchtime", "1x")
+	flag.Parse()
+
+	commitResults := BenchmarkCommitResults{
+		Commit:  commit,
+		Results: make(map[string]BenchmarkResult)}
+
+	for _, benchmarkFunc := range benchmarkFuncs {
+		benchmarkResult := BenchmarkResult{
+			TestName:      GetFunctionName(benchmarkFunc),
+			NumberOfTests: testNum,
+			testFunction:  benchmarkFunc}
+
+		for i := 0; i < benchmarkResult.NumberOfTests; i++ {
+			// fmt.Printf("TestName: %s\n", benchmarkResult.TestName)
+			res := testing.Benchmark(benchmarkResult.testFunction)
+			// fmt.Printf("%d: %f\n", i, res.T.Seconds())
+			benchmarkResult.addResult(res)
+		}
+		benchmarkResult.updateStats()
+
+		commitResults.Results[benchmarkResult.TestName] = benchmarkResult
+	}
+
+	json, _ := json.MarshalIndent(commitResults, "", " ")
+	fmt.Print(string(json[:]))
+}
+
+type BenchmarkCommitResults struct {
+	Commit  string                     `json:"commit"`
+	Results map[string]BenchmarkResult `json:"benchmarkResults"`
+}
+
+type BenchmarkResult struct {
+	TestName      string `json:"testName"`
+	NumberOfTests int    `json:"numberOfTests"`
+	testFunction  func(*testing.B)
+	TestsRun      int       `json:"testsRun"`
+	testTimes     []float64 `json:"-"`
+	StdDev        float64   `json:"stdDev"`
+	Mean          float64   `json:"mean"`
+	Min           float64   `json:"min"`
+	Pct25         float64   `json:"pct25"`
+	Pct50         float64   `json:"pct50"`
+	Pct75         float64   `json:"pct75"`
+	Pct90         float64   `json:"pct90"`
+	Max           float64   `json:"max"`
+}
+
+// add a single result to testTimes
+func (driver *BenchmarkResult) addResult(individualResult testing.BenchmarkResult) {
+	//TODO add assert for single test
+	driver.TestsRun = driver.TestsRun + individualResult.N
+	driver.testTimes = append(driver.testTimes, individualResult.T.Seconds())
+}
+
+// calculate statistical metrics for all testTimes
+func (results *BenchmarkResult) updateStats() {
+	//TODO add assert for single test
+	results.StdDev, _ = stats.StandardDeviation(results.testTimes)
+	results.Mean, _ = stats.Mean(results.testTimes)
+	results.Min, _ = stats.Min(results.testTimes)
+	results.Pct25, _ = stats.Percentile(results.testTimes, 25)
+	results.Pct50, _ = stats.Percentile(results.testTimes, 50)
+	results.Pct75, _ = stats.Percentile(results.testTimes, 75)
+	results.Pct90, _ = stats.Percentile(results.testTimes, 90)
+	results.Max, _ = stats.Max(results.testTimes)
+}
+
+func BenchmarkRandom(b *testing.B) {
+	for i := 0; i < b.N; i++ {
+		rand.Seed(time.Now().UnixNano())
+		sleepTime := time.Duration(rand.Int63n(50))
+		// fmt.Printf("SleepTime = %d\n", sleepTime)
+		time.Sleep(sleepTime * time.Second / 100)
+	}
+}
+
+// This is a bad test. client.Fetch runtime depends on network speed.
+func BenchmarkGetImage(b *testing.B) {
+	client, err := newClient(b, address)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error new client : %s\n", err)
```

Comment:
You can use b's methods instead of fmt.Fprintf

Reply:
I will look into this. The docs in question refer to an error log but don't describe what it is or how it works. I need to determine if it can [also] be shown to stderr.